de.metas.report/report-service/src/main/java/de/metas/adempiere/report/jasper/JsonDataSourceService.java

```diff
@@ -94,6 +95,29 @@ public InputStream getInputStream(@NonNull final ReportContext reportContext) th
 		return is;
 	}
 
+	private UserAuthToken getUserAuhToken()
+	{
+		final UserId userId = Env.getLoggedUserId();
+		final RoleId roleId = Env.getLoggedRoleId();
+
+		UserAuthToken token = userAuthTokenRepo.retrieveByUserIdOrNull(userId, roleId);
+
+		//
+		// fallback to default Jasper Json user
+		if (token == null)
+		{
+			final ClientId clientId = Env.getClientId();
+			Services.get(IUserDAO.class).retrieveUserIdByValue(JSONREPORTS_USER_VALUE, clientId);
```

Comment:
As written further up, if there is no better solution (that is not falling back to the "can-do-anything-rolle"), pls specify a dedicated role.
Also, I'm not sure if we want a fallback after all. There might be a token for the currently logged in user and role, but it migh be "too weak" for reporting. So maybe better don't fall back, but use the dedicated user&role in the first place?